File: src/main/java/cn/zjiali/robot/model/response/RobotBaseResponse.java
package cn.zjiali.robot.model.response;

/**
 * @author zJiaLi
 * @since 2021-05-08 10:02
 */
public class RobotBaseResponse <T>{

    private int status;

    private String message;

    private T data;

    public RobotBaseResponse() {
    }

    public RobotBaseResponse(int status, String message, T data) {
        this.status = status;
        this.message = message;
        this.data = data;
    }

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }
}

File: src/main/java/cn/zjiali/robot/model/response/RobotBaseResponse.kt
package cn.zjiali.robot.model.response

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * @author zJiaLi
 * @since 2021-05-08 10:02
 */
@Serializable
class RobotBaseResponse<T> {
    @SerialName(value = "status")
    var status = 0

    @SerialName(value = "message")
    var message: String? = null

    @SerialName(value = "data")
    var data: T? = null
        private set

    constructor()
    constructor(status: Int, message: String?, data: T) {
        this.status = status
        this.message = message
        this.data = data
    }

    fun success(): Boolean {
        return status == 200
    }

    fun setData(data: T) {
        this.data = data
    }
}